### PR TITLE
Refs #29144 - Use systemd socket activation

### PR DESCRIPTION
**Pull request by ekohl**
This gives reliable service restarts since systemd keeps the socket open.

Replacement of ExecStart is no longer needed since foreman core now calls /usr/share/foreman/bin/rails which is already modified to use /usr/bin/tfm-ruby.

It also correcs the macros to only execute on the subpackage. The incorrect systemd_postun_with_restart (systemctl try-restart) is changed to the appropriate systemd_post (systemctl preset).

Depends on https://github.com/theforeman/foreman/pull/7536

### CODE REVIEW

**Comment by ehelms** [comment]
Looks good, might as well bump the release.

**Comment by ehelms** [comment]
[test rpm]

**Comment by ehelms** [comment]
[test rpm]

File: packages/foreman/foreman/foreman.spec [hunk 1]
```diff
@@ -771,7 +772,6 @@ install -Dp -m0644 %{SOURCE5} %{buildroot}%{_tmpfilesdir}/%{name}.conf
 install -Dp -m0644 extras/systemd/%{name}.service %{buildroot}%{_unitdir}/%{name}.service
```
Comment (ehelms) [inline packages/foreman/foreman/foreman.spec]: Need to install the socket as well here

**Comment by ekohl** [comment]
Thanks, updated and now green.

**Comment by ekohl** [comment]
Also updated now to drop the unused sysconfig file.

**Pull request merged**